### PR TITLE
Adopt scalafmt for Scala code style

### PR DESCRIPTION
**Pull request by Baltoli**
In https://github.com/runtimeverification/k/pull/3841, we adopted the `scalafmt` tool to auto-format the Scala code in the K frontend. This PR adopts a precisely identical formatting setup for the LLVM backend's Scala code.

### CODE REVIEW

**Comment by Baltoli** [comment]
Blocked on the maven dependency update job being out of date: #921

**Review by Robertorosmaninho — APPROVED** [review]
Overall LGTM! I would expect @Scott-Guest to approvals for a more precise review of the config files!

File: matching/pom.xml [hunk 1]
```diff
@@ -31,6 +31,14 @@
...
   </repositories>
 
+  <properties>
+    <scala.majorVersion>2.12</scala.majorVersion>
+    <scala.minorVersion>18</scala.minorVersion>
+    <scala.version>${scala.majorVersion}.${scala.minorVersion}</scala.version>
+    <spotless.version>2.41.1</spotless.version>
+    <scalafmt.version>3.7.17</scalafmt.version>
+  </properties>
+
```
Comment (Scott-Guest) [inline matching/pom.xml:41]: As far as I understand, all the properties are automatically inherited from the parent POM, so we don't need to redeclare them here.

File: matching/pom.xml [hunk 2]
```diff
@@ -59,6 +67,16 @@
...
+    <dependency>
+      <groupId>com.diffplug.spotless</groupId>
+      <artifactId>spotless-maven-plugin</artifactId>
+      <version>${spotless.version}</version>
+    </dependency>
+    <dependency>
+      <groupId>org.scalameta</groupId>
+      <artifactId>scalafmt-core_${scala.majorVersion}</artifactId>
+      <version>${scalafmt.version}</version>
+    </dependency>
```
Comment (Scott-Guest) [inline matching/pom.xml:79]: Same thing here - dependencies will automatically be pulled in from the parent POM and don't need to be redeclared

**Review by Scott-Guest — APPROVED** [review]
LGTM overall, but see my comments about the property and dependency declarations

**Review by gtrepta — APPROVED** [review]

**Comment by Baltoli** [comment]
Thanks @Scott-Guest - I didn't realise Maven worked like that!

**Comment by Baltoli** [comment]
I have addressed the duplicated configuration elements; going to merge this now.

**Comment by Scott-Guest** [comment]
@Baltoli Hmm, I am pretty sure this is how Maven works, and it passes locally for me when removing the re-declarations. The Maven documentation explicitly states that properties and dependencies are inherited: https://maven.apache.org/pom.html#inheritance

I see that CI is failing due to the undefined properties, so I might be missing something though!

**Comment by Baltoli** [comment]
I seem to have found the correct incantation to get Maven to inherit the properties and configure the plugin now!

**Pull request merged**